store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -48,25 +54,26 @@
  */
 @Since("2.0.0")
 public class FederatedOperationHandler<INPUT, OUTPUT> implements OperationHandler<FederatedOperation<INPUT, OUTPUT>> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FederatedOperationHandler.class);
 
     public static final String ERROR_WHILE_RUNNING_OPERATION_ON_GRAPHS_FORMAT = "Error while running operation on graphs, due to: %s";
-    public static final String MERGE_FUNCTION_GET_GRAPHS = FederatedOperationHandler.class.getCanonicalName() + "merge.function.getGraphs";
+    private List<GraphSerialisable> graphs;
     private Context context;
 
     @Override
     public Object doOperation(final FederatedOperation<INPUT, OUTPUT> operation, final Context context, final Store store) throws OperationException {
         this.context = context;
-        final Iterable<?> allGraphResults = getAllGraphResults(operation, (FederatedStore) store);
+        this.graphs = getGraphs(operation, context, (FederatedStore) store);
+        final Iterable<?> allGraphResults = getAllGraphResults(operation);
 
         return mergeResults(allGraphResults, operation, (FederatedStore) store);
     }
 
-    private Iterable getAllGraphResults(final FederatedOperation<INPUT, OUTPUT> operation, final FederatedStore store) throws OperationException {
+    private Iterable getAllGraphResults(final FederatedOperation<INPUT, OUTPUT> operation) throws OperationException {
         try {
             List<Object> results;
-            final Collection<GraphSerialisable> graphs = getGraphs(operation, context, store);
-            context.setVariable(MERGE_FUNCTION_GET_GRAPHS, graphs.stream().map(GraphSerialisable::getGraphId).collect(Collectors.toList()));
```

Comment:
Information of graphs already existed and was retained here. This one line could have been modified to store the whole GraphSerialisable.